feat: adding swagger

maxquinones opened the pull request:
## Description
<!-- Provide a brief description of the changes in this PR -->
This is PR adds in the Swagger meta data needed to create the OAS file with the description and details for documentation purposes on the SEMOSS documentation site. 

The OAS file is created at build time.

<img width="512" height="301" alt="image" src="https://github.com/user-attachments/assets/e327f04c-ceba-4f9e-9457-a53dec9f913d" />


## Changes Made
<!-- List the key changes made in this PR -->
Added changes to the javax-rs
Two files are OAS files are generated one is a Yaml file and the other is  a JSON file. Either format can be used in the documentation app. 

## How to Test
<!-- Explain how reviewers can test your changes -->
1. Steps to reproduce/test the behavior
3. Expected outcomes

## Notes
<!-- Any further notes regarding changes -->

maxquinones committed:
feat: adding swagger

QodoAI-Agent commented:
## Title

draft: feat: adding swagger

___
### **User description**
## Description
<!-- Provide a brief description of the changes in this PR -->

## Changes Made
<!-- List the key changes made in this PR -->

## How to Test
<!-- Explain how reviewers can test your changes -->
1. Steps to reproduce/test the behavior
2. Expected outcomes

## Notes
<!-- Any further notes regarding changes -->


___

### **PR Type**
Enhancement, Documentation


___

### **Description**
- Add Swagger Maven plugin for API docs

- Configure sources, host, base path

- Generate JSON and YAML outputs

- Bind generation to compile phase


___

### Diagram Walkthrough


```mermaid
flowchart LR
  POM["Update pom.xml build plugins"] -- "add" --> SWAGGER["Swagger Maven Plugin 3.1.8"]
  SWAGGER -- "configured with" --> CFG["locations, schemes, host, basePath, info"]
  SWAGGER -- "outputs" --> ART["Swagger JSON/YAML in target/swagger-ui"]
  SWAGGER -- "bound to" --> PHASE["compile phase (generate)"]
```



<details> <summary><h3> File Walkthrough</h3></summary>

<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>pom.xml</strong><dd><code>Introduce Swagger generation via Maven plugin</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pom.xml

<ul><li>Add <code>swagger-maven-plugin</code> 3.1.8 under build plugins<br> <li> Configure API source: <code>prerna.semoss.web.services.local*</code>, http/https, <br>host, base path<br> <li> Set API info: title, version, description<br> <li> Output Swagger JSON/YAML to <code>${project.build.directory}/swagger-ui</code>; run <br>at compile</ul>


</details>


  </td>
  <td><a href="https://github.com/SEMOSS/Monolith/pull/168/files#diff-9c5fb3d1b7e3b0f54bc5c4182965c4fe1f9023d449017cece3005d3f90e8e4d8">+40/-0</a>&nbsp; &nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

</details>

___

QodoAI-Agent commented:
## PR Reviewer Guide 🔍

Here are some key observations to aid the review process:

<table>
<tr><td>⏱️&nbsp;<strong>Estimated effort to review</strong>: 2 🔵🔵⚪⚪⚪</td></tr>
<tr><td>🧪&nbsp;<strong>No relevant tests</strong></td></tr>
<tr><td>🔒&nbsp;<strong>No security concerns identified</strong></td></tr>
<tr><td>⚡&nbsp;<strong>Recommended focus areas for review</strong><br><br>

<details><summary><a href='https://github.com/SEMOSS/Monolith/pull/168/files#diff-9c5fb3d1b7e3b0f54bc5c4182965c4fe1f9023d449017cece3005d3f90e8e4d8R130-R133'><strong>Plugin Version</strong></a>

The added swagger-maven-plugin uses an older 3.1.8 version; verify compatibility with your Maven/Java stack and consider updating to a maintained version to receive fixes and OpenAPI improvements.
</summary>

```xml
<groupId>com.github.kongchen</groupId>
<artifactId>swagger-maven-plugin</artifactId>
<version>3.1.8</version>
<configuration>
```

</details>

<details><summary><a href='https://github.com/SEMOSS/Monolith/pull/168/files#diff-9c5fb3d1b7e3b0f54bc5c4182965c4fe1f9023d449017cece3005d3f90e8e4d8R137-R139'><strong>Package Scan Pattern</strong></a>

The locations pattern 'prerna.semoss.web.services.local*' may miss controllers outside this package or unintentionally include unintended classes; confirm it matches your actual REST resources.
</summary>

```xml
<locations>
	<location>prerna.semoss.web.services.local*</location>
</locations>
```

</details>

<details><summary><a href='https://github.com/SEMOSS/Monolith/pull/168/files#diff-9c5fb3d1b7e3b0f54bc5c4182965c4fe1f9023d449017cece3005d3f90e8e4d8R144-R146'><strong>Hardcoded Host/BasePath</strong></a>

Host 'localhost:8080' and basePath '/api' are hardcoded; this can produce misleading specs across environments. Consider parameterizing via Maven profiles/properties.
</summary>

```xml
<host>localhost:8080</host>
<basePath>/api</basePath>
<info>
```

</details>

</td></tr>
</table>

QodoAI-Agent commented:
## PR Code Suggestions ✨
<!-- 5a02684 -->

Latest suggestions up to 5a02684
<table><thead><tr><td><strong>Category</strong></td><td align=left><strong>Suggestion&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </strong></td><td align=center><strong>Impact</strong></td></tr><tbody><tr><td rowspan=5>Possible issue</td>
<td>



<details><summary>Restore lost admin utils assignment</summary>

___


**The result of <code>performAdminCheck</code> is no longer assigned to <code>adminUtils</code>, which is later <br>used in this method and others. This will cause a NullPointerException or logic <br>failure when accessing admin utilities. Restore the assignment to ensure admin <br>checks return the utility instance used below.**

[src/prerna/semoss/web/services/local/auth/AdminProjectAuthorizationResource.java [103-107]](https://github.com/SEMOSS/Monolith/pull/168/files#diff-ebd5f88af3d76ef47fa3b7d553e69593b3dda415c6d79a54c4af7278c420954fR103-R107)

```diff
 user = ResourceUtility.getUser(request);
-performAdminCheck(request, user);
+adminUtils = performAdminCheck(request, user);
```
<details><summary>Suggestion importance[1-10]: 8</summary>

__

Why: The PR removed assigning the result of `performAdminCheck` to `adminUtils` but `adminUtils` is used later, risking a NullPointerException; restoring the assignment is accurate and critical for functionality.


</details></details></td><td align=center>Medium

</td></tr><tr><td>



<details><summary>Use typed deserialization for lists</summary>

___


**Using raw List.class for Gson deserialization loses generic type information and can <br>yield malformed element types at runtime. Use a TypeToken to deserialize into a <br>proper List<String>.**

[src/prerna/semoss/web/services/local/auth/DatabaseAuthorizationResource2.java [663]](https://github.com/SEMOSS/Monolith/pull/168/files#diff-7fcd54375f7a939268fcd9223bc918eb1a139ac02ba1c90a1dcb870885262309R663-R663)

```diff
-List<String> ids = gson.fromJson(form.getFirst("ids"), List.class);
+Type idsType = new TypeToken<List<String>>(){}.getType();
+List<String> ids = gson.fromJson(form.getFirst("ids"), idsType);
```
<details><summary>Suggestion importance[1-10]: 7</summary>

__

Why: The PR already introduces `TypeToken` patterns elsewhere; switching from `List.class` to a `TypeToken<List<String>>` prevents raw type deserialization issues and is accurate and beneficial.


</details></details></td><td align=center>Medium

</td></tr><tr><td>



<details><summary>Validate bulk permission payload</summary>

___


**Guard against null or malformed <code>userpermissions</code> to avoid <code>NullPointerException</code> and <br>invalid casts. Validate the parsed list before use and return a 400 with a helpful <br>message if it's missing or not an array.**

[src/prerna/semoss/web/services/local/auth/EngineAuthorizationResource.java [515-516]](https://github.com/SEMOSS/Monolith/pull/168/files#diff-5f63dd8a2b478a8295a1f52a93fc65b739d114355953e81ba4b2964afd1d4ad0R515-R516)

```diff
 Type listMapObj = new TypeToken<List<Map<String, Object>>>(){}.getType();
-List<Map<String, Object>> permission = new Gson().fromJson(form.getFirst("userpermissions"), listMapObj);
+String rawPermissions = form.getFirst("userpermissions");
+if (rawPermissions == null) {
+    Map<String, Object> error = new HashMap<>();
+    error.put(Constants.ERROR_MESSAGE, "Missing required parameter: userpermissions");
+    return WebUtility.getResponse(error, 400);
+}
+List<Map<String, Object>> permission = new Gson().fromJson(rawPermissions, listMapObj);
+if (permission == null) {
+    Map<String, Object> error = new HashMap<>();
+    error.put(Constants.ERROR_MESSAGE, "Invalid format for userpermissions. Expected JSON array.");
+    return WebUtility.getResponse(error, 400);
+}
```
<details><summary>Suggestion importance[1-10]: 7</summary>

__

Why: Adding null/format validation for `userpermissions` prevents NPEs and gives clients clear feedback; it's accurate and improves robustness without altering logic.


</details></details></td><td align=center>Medium

</td></tr><tr><td>



<details><summary>Validate IDs array before use</summary>

___


**Check for missing or invalid <code>ids</code> before sanitization to prevent null dereference and <br>unclear errors. Return a 400 with a clear message when <code>ids</code> is absent or not an <br>array.**

[src/prerna/semoss/web/services/local/auth/AdminEngineAuthorizationResource.java [1020-1022]](https://github.com/SEMOSS/Monolith/pull/168/files#diff-0f0933730b4645b54e156f61b0fff9875e50a6cbd7de84f2eb87490c16a1ea39R1020-R1022)

```diff
 Type listOfString = new TypeToken<List<String>>(){}.getType();
-List<String> ids = gson.fromJson(form.getFirst("ids"), listOfString);
+String rawIds = form.getFirst("ids");
+if (rawIds == null) {
+    Map<String, Object> error = new HashMap<>();
+    error.put(Constants.ERROR_MESSAGE, "Missing required parameter: ids");
+    return WebUtility.getResponse(error, 400);
+}
+List<String> ids = gson.fromJson(rawIds, listOfString);
+if (ids == null) {
+    Map<String, Object> error = new HashMap<>();
+    error.put(Constants.ERROR_MESSAGE, "Invalid format for ids. Expected JSON array of strings.");
+    return WebUtility.getResponse(error, 400);
+}
 ids = WebUtility.inputSQLSanitizer(ids);
```
<details><summary>Suggestion importance[1-10]: 7</summary>

__

Why: Checking `ids` for presence and valid JSON before sanitization avoids null dereferences and improves error reporting; matches the changed code context.


</details></details></td><td align=center>Medium

</td></tr><tr><td>



<details><summary>Fix response schema mismatch</summary>

___


**The OpenAPI annotation declares an array response but the method returns an object <br>structure, not a bare array. This mismatch can break client generation and API docs. <br>Update the schema to reflect the actual JSON shape returned.**

[src/prerna/semoss/web/services/local/auth/DatabaseAuthorizationResource2.java [69-73]](https://github.com/SEMOSS/Monolith/pull/168/files#diff-7fcd54375f7a939268fcd9223bc918eb1a139ac02ba1c90a1dcb870885262309R69-R73)

```diff
 @ApiResponse(responseCode = "200", description = "Databases retrieved",
             content = @Content(mediaType = "application/json",
-                array = @ArraySchema(schema = @Schema(type = "object"))
+                schema = @Schema(type = "object")
             )
         ),
```
<details><summary>Suggestion importance[1-10]: 6</summary>

__

Why: The method returns an object (not a bare array), so documenting it as an array can mislead clients; updating to `schema = @Schema(type = "object")` improves accuracy, though it's a docs fix, not a critical bug.


</details></details></td><td align=center>Low

</td></tr><tr><td rowspan=3>General</td>
<td>



<details><summary>Correct forbidden status documentation</summary>

___


**Several endpoints return 401 for authorization failures (admin required), which <br>should be 403 to correctly indicate authenticated but forbidden access. Adjust the <br>OpenAPI docs to 403 where applicable to avoid misleading clients.**

[src/prerna/semoss/web/services/local/auth/DatabaseAuthorizationResource2.java [351-356]](https://github.com/SEMOSS/Monolith/pull/168/files#diff-7fcd54375f7a939268fcd9223bc918eb1a139ac02ba1c90a1dcb870885262309R351-R356)

```diff
-@ApiResponse(responseCode = "401", description = "Unauthorized or admin required",
+@ApiResponse(responseCode = "403", description = "Forbidden (admin required)",
 			content = @Content(mediaType = "application/json", schema = @Schema(type = "object"))
 		),
```
<details><summary>Suggestion importance[1-10]: 5</summary>

__

Why: Updating 401 to 403 for admin-required failures improves API accuracy, but it's a documentation change; correctness depends on actual runtime responses, which aren't changed here.

</details></details></td><td align=center>Low

</td></tr><tr><td>



<details><summary>Avoid duplicate security scheme declaration</summary>

___


**Ensure the security scheme annotation is only declared once at application level to <br>avoid duplicate OpenAPI components. If kept here, move <code>@SecurityScheme</code> to a central <br>configuration class and leave only <code>@SecurityRequirement</code> on the resource.**

[src/prerna/semoss/web/services/local/ProjectResource.java [92-94]](https://github.com/SEMOSS/Monolith/pull/168/files#diff-eda39d59d6984b3018af379f78509b936ba4ce35f4bea9c1d07fc2bfc61fb488R92-R94)

```diff
 @SecurityRequirement(name = "basicAuth")
-@SecurityScheme(name = "basicAuth", type = SecuritySchemeType.HTTP, scheme = "basic")
```
<details><summary>Suggestion importance[1-10]: 5</summary>

__

Why: Centralizing `@SecurityScheme` can reduce duplicated OpenAPI components, but it's an organizational concern and may depend on project conventions; moderate impact.

</details></details></td><td align=center>Low

</td></tr><tr><td>



<details><summary>Align form key with list parsing</summary>

___


**The form field name suggests a single <code>insightId</code>, but the code parses a list, which <br>can break clients sending a scalar. Either change the form key to <code>insightIds</code> or <br>accept both scalar and JSON array inputs to avoid runtime parsing errors. Aligning <br>the key with list intent is the least invasive.**

[src/prerna/semoss/web/services/local/auth/AdminInsightAuthorizationResource.java [226-228]](https://github.com/SEMOSS/Monolith/pull/168/files#diff-689d01dbaf0b0678082a58d883f94935ee4cc30363ca6851064c5856c4f37bd3R226-R228)

```diff
 Gson gson = new Gson();
 Type insightIdsType = new TypeToken<List<String>>(){}.getType();
-List<String> insightIds = gson.fromJson(form.getFirst("insightId"), insightIdsType);
+String raw = form.getFirst("insightIds"); // expect plural for lists
+List<String> insightIds = gson.fromJson(raw, insightIdsType);
```
<details><summary>Suggestion importance[1-10]: 4</summary>

__

Why: The observation about singular key versus list parsing is reasonable, but changing the form field to `insightIds` is a behavioral API change not reflected in the PR context; a more robust approach would accept both formats.


</details></details></td><td align=center>Low

</td></tr><tr><td rowspan=1>Security</td>
<td>



<details><summary>Resolve conflicting security annotations</summary>

___


**Applying <code>@SecurityRequirement</code> while also using <code>@PermitAll</code> is contradictory and may <br>expose endpoints that Swagger marks as secured. Remove <code>@PermitAll</code> or narrow it <br>per-method to match the intended security posture and avoid unintentionally open <br>endpoints.**

[src/prerna/semoss/web/services/local/NameServer.java [124-129]](https://github.com/SEMOSS/Monolith/pull/168/files#diff-4383d399a72d955ae750e7a2fd69d68424da9be8e872c9c453cc6ccc4cc5be56R124-R129)

```diff
 @SecurityRequirement(name = "basicAuth")
-@PermitAll
+// Removed @PermitAll to avoid conflicting open access
```
<details><summary>Suggestion importance[1-10]: 3</summary>

__

Why: While noting potential ambiguity between `@SecurityRequirement("basicAuth")` and `@PermitAll` is valid, simply removing `@PermitAll` may not match the intended access; the suggestion lacks context and could introduce breaking access changes.

</details></details></td><td align=center>Low

</td></tr></tr></tbody></table>

___

#### Previous suggestions
<details><summary>Suggestions up to commit 261f023</summary>
<br><table><thead><tr><td><strong>Category</strong></td><td align=left><strong>Suggestion&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </strong></td><td align=center><strong>Impact</strong></td></tr><tbody><tr><td rowspan=3>General</td>
<td>



<details><summary>Use safer output directory</summary>

___


**Generating into the build directory risks being wiped during clean and may clash <br>with packaging. Output to a classifier-specific directory or a dedicated folder <br>under target to avoid conflicts.**

[pom.xml [154]](https://github.com/SEMOSS/Monolith/pull/168/files#diff-9c5fb3d1b7e3b0f54bc5c4182965c4fe1f9023d449017cece3005d3f90e8e4d8R154-R154)

```diff
-<swaggerDirectory>${project.build.directory}/swagger-ui</swaggerDirectory>
+<swaggerDirectory>${project.build.directory}/generated-resources/swagger</swaggerDirectory>
```
<details><summary>Suggestion importance[1-10]: 7</summary>

__

Why: Directing Swagger output to `generated-resources/swagger` follows Maven conventions and reduces packaging conflicts; it's accurate and beneficial for build hygiene.

</details></details></td><td align=center>Medium

</td></tr><tr><td>



<details><summary>Externalize host configuration</summary>

___


**Avoid hardcoding the host to prevent incorrect URLs when building in different <br>environments. Make it configurable via a Maven property with a default.**

[pom.xml [144]](https://github.com/SEMOSS/Monolith/pull/168/files#diff-9c5fb3d1b7e3b0f54bc5c4182965c4fe1f9023d449017cece3005d3f90e8e4d8R144-R144)

```diff
-<host>localhost:8080</host>
+<host>${swagger.host}</host>
```
<details><summary>Suggestion importance[1-10]: 6</summary>

__

Why: Making `<host>` configurable improves portability across environments and aligns with Maven best practices; the change is accurate and low-risk but not critical.


</details></details></td><td align=center>Low

</td></tr><tr><td>



<details><summary>Parameterize base path</summary>

___


**Do not hardcode the base path, as it may differ across deployments. Use a Maven <br>property with a sensible default to avoid broken Swagger links.**

[pom.xml [145]](https://github.com/SEMOSS/Monolith/pull/168/files#diff-9c5fb3d1b7e3b0f54bc5c4182965c4fe1f9023d449017cece3005d3f90e8e4d8R145-R145)

```diff
-<basePath>/api</basePath>
+<basePath>${swagger.basePath}</basePath>
```
<details><summary>Suggestion importance[1-10]: 5</summary>

__

Why: Externalizing `<basePath>` can help in multi-environment setups; it's reasonable and correct but has moderate impact since many apps keep a stable base path.


</details></details></td><td align=center>Low

</td></tr></tr></tbody></table>

</details>

maxquinones committed:
updating swagger  configuration

maxquinones committed:
swagger annotations

maxquinones committed:
adding swagger annotations

maxquinones committed:
adding annotations for Name server

maxquinones committed:
updates to name server

maxquinones committed:
update swagger documentation for API and Scheduler

maxquinones committed:
add Swagger annotations for Authorization and NameServer APIs

maxquinones committed:
update Swagger annotations for NameServer API

maxquinones committed:
update Swagger documentation for API and Authorization endpoints

maxquinones committed:
add Swagger annotations and tags for AI Models, Session, and Authentication APIs

maxquinones committed:
update Swagger documentation for LegacyAppResource and related APIs

maxquinones committed:
Enhance AdminProjectAuthorizationResource with OpenAPI Annotations

- Added OpenAPI annotations to document endpoints for project authorization management.
- Included detailed operation summaries and response descriptions for better API documentation.
- Updated JSON parsing to use TypeToken for improved type safety.
- Enhanced error handling responses for various endpoints, including unauthorized and server error cases.

maxquinones committed:
Enhance API documentation and improve response handling

- Added OpenAPI annotations to EngineAuthorizationResource, GroupEngineAuthorizationResource, GroupInsightAuthorizationResource, GroupProjectAuthorizationResource, and InsightAuthorizationResource for better API documentation.
- Improved response handling by using TypeToken for JSON deserialization in multiple methods across the aforementioned resources.
- Introduced LegacyAdminSuccessResponse class to standardize success responses for legacy admin endpoints.

maxquinones committed:
Update Swagger documentation for API endpoints and improve descriptions

maxquinones committed:
Update Swagger documentation for Engine Route and Database Engine APIs

maxquinones committed:
Update Swagger documentation for MCP and related APIs

maxquinones committed:
Update Swagger documentation for MCP and related APIs

maxquinones committed:
Update Swagger documentation for MCP and Model Engine APIs

maxquinones committed:
Update Swagger documentation for API endpoints and improve descriptions

maxquinones committed:
Update Swagger documentation for Project endpoints, adding descriptions and responses

maxquinones committed:
Update Swagger documentation for Project endpoints, adding descriptions and responses

maxquinones committed:
Merge branch 'dev' of github.com-semoss:SEMOSS/Monolith into swagger-feature

maxquinones committed:
feat: pass logging context to MCPReaper for enhanced logging capabilities

maxquinones committed:
chore: update swagger documentation for consistency and clarity

maxquinones committed:
refactor: update API title from SEMOSS to AI-Core in swagger configuration files

maxquinones committed:
Merge branch 'dev' of github.com-semoss:SEMOSS/Monolith into swagger-feature

maxquinones committed:
chore: update project dependencies and configuration

maxquinones committed:
chore: update API title in swagger configuration files

roxyramos commented on WebContent/WEB-INF/web.xml:
```diff
@@ -681,7 +681,7 @@
    <context-param>
       <description>The place where the core RDF Map gets picked from</description>
       <param-name>RDF-MAP</param-name>
-      <param-value>C:/workspace/Semoss/RDF_Map.prop</param-value>
```
@maxquinones this change is not needed

maxquinones replied:
done reverted change

maxquinones committed:
fix: update RDF map path to use forward slashes for compatibility

maxquinones committed:
Merge branch 'dev' of github.com-semoss:SEMOSS/Monolith into swagger-feature

maxquinones committed:
chore: update swagger configuration files for consistency

maxquinones committed:
Merge branch 'dev' of github.com-semoss:SEMOSS/Monolith into swagger-feature

maxquinones committed:
Merge branch 'dev' of github.com-semoss:SEMOSS/Monolith into swagger-feature

maxquinones committed:
Merge branch 'dev' of github.com-semoss:SEMOSS/Monolith into swagger-feature
